data/gen.go

```diff
@@ -100,6 +100,7 @@ func createCodeOutput(s string) {
 	}
 	_, err = file.WriteString("// Code generated - DO NOT EDIT.\n" +
 		"// Regenerate with `go generate`.\n\n" +
+		"// go:generate go run data/gen.go\n" +
```

Comment:
This should remain in pokedex.go, because `go generate` should work without the presence of the generated code.

You gotta start putting these things in separate PRs. It makes it really hard to review your PRs, and a few pretty significant bugs have gotten through as a result. Making refactors when implementing a feature usually means that you don't have what is required to implement the feature, and chances are somebody else needs it too. Make the refactor in a different PR, stating your reason, and then rebase your feature's branch on main when it gets merged in.